From HEAD Mon Sep 17 00:00:00 2001
From: Hypothesis 6.139.2 <no-reply@hypothesis.works>
Date: Tue, 13 Jan 2026 02:05:12
Subject: [PATCH] Hypothesis: add explicit examples

---
--- ./backend/photon_algebra/tests/test_bounded_confluence.py
+++ ./backend/photon_algebra/tests/test_bounded_confluence.py
@@ -130,6 +130,9 @@
 
 @settings(max_examples=250, deadline=None)
 @given(exprs)
+@example(
+    e={'op': '⊗', 'states': ['a', 'b', {'op': '⊕', 'states': ['a', 'b']}]},
+).via('discovered failure')
 def test_bounded_confluence_via_equivalent_variants(e):
     nf = rewriter.normalize(deepclone(e))
     for v in variants(e):
--- ./backend/photon_algebra/tests/test_properties_normalize.py
+++ ./backend/photon_algebra/tests/test_properties_normalize.py
@@ -92,12 +92,49 @@
 
 @settings(max_examples=200, deadline=None)
 @given(exprs, exprs)
+@example(
+    a={'op': '⊕',
+     'states': [{'op': '⊗', 'states': []}, {'op': '⊕', 'states': ['b']}]},
+    b={'op': '⊕',
+     'states': [{'op': '⊕', 'states': []},
+      {'op': '⊕', 'states': []},
+      'a',
+      {'op': '⊕',
+       'states': ['a',
+        {'op': '⊕', 'states': ['b', 'd']},
+        {'op': '⊕', 'states': ['d', {'op': '∅'}, 'b']},
+        {'op': '⊗', 'states': ['b', {'op': '∅'}, 'a', {'op': '∅'}]}]}]},
+).via('discovered failure')
 def test_commutativity_stability_product(a, b):
     assert rewriter.normalize(deepclone(PROD(a, b))) == rewriter.normalize(deepclone(PROD(b, a)))
 
 
 @settings(max_examples=200, deadline=None)
 @given(exprs, exprs, exprs)
+@example(
+    a={'op': '⊗',
+     'states': ['b',
+      {'op': '⊕', 'states': [{'op': '∅'}]},
+      {'op': '⊗',
+       'states': [{'op': '⊕', 'states': []},
+        {'op': '⊗',
+         'states': [{'op': '⊕', 'states': ['d']},
+          {'op': '∅'},
+          {'op': '⊗',
+           'states': [{'op': '∅'}, {'op': '∅'}, {'op': '∅'}, {'op': '∅'}]},
+          {'op': '⊗', 'states': [{'op': '∅'}, 'c']}]},
+        {'op': '⊗',
+         'states': ['c',
+          {'op': '⊕', 'states': ['a', {'op': '∅'}, 'b', {'op': '∅'}]}]},
+        {'op': '⊗',
+         'states': [{'op': '∅'},
+          {'op': '⊕', 'states': [{'op': '∅'}, 'a', 'a']}]}]},
+      {'op': '⊕', 'states': []}]},
+    b={'op': '⊕',
+     'states': [{'op': '⊕', 'states': [{'op': '∅'}]},
+      {'op': '⊗', 'states': ['c', 'd', 'c']}]},
+    c={'op': '⊗', 'states': []},
+).via('discovered failure')
 def test_associativity_stability_sum(a, b, c):
     left = SUM(SUM(a, b), c)
     right = SUM(a, SUM(b, c))
